packages/cli/src/events/WorkflowStatistics.ts

```diff
@@ -72,16 +72,20 @@ export async function nodeFetchedData(workflowId: string, node: INode): Promise<
 		return;
 	}
 
-	// Update only if necessary
-	const response = await Db.collections.Workflow.update(
-		{ id, dataLoaded: false },
-		{ dataLoaded: true },
-	);
-
-	// If response.affected is 1 then we know this was the first time data was loaded into the workflow; do posthog event here
-	if (!response.affected) return;
+	// Try to insert the data loaded statistic
+	try {
+		await Db.collections.WorkflowStatistics.insert({
+			workflowId: id,
+			name: StatisticsNames.dataLoaded,
+			count: 1,
+			latestEvent: new Date(),
+		});
+	} catch (error) {
+		// If this fails, it'll be a duplicate key failure. It is safe to return from here
```

Comment:
this could also be a network error. assuming that this is going to be a constraint error without checking for it is only going to cause another issue in the future. We should add some kind of assertion on the error object, and re-throw any unexpected errors.